retry/retry.go

```diff
@@ -11,27 +11,31 @@ import (
 	"github.com/pkg/errors"
 )
 
-const taskSystemName = "task"
+const (
+	taskSystemName  = "task"
+	maxWaitTimeInNs = 15 * time.Minute // 15 minutes
```

Comment:
The switch to time.Duration is clean! Wanted to double check if it's still necessary to have `InNs` as part of this constant name? Similarly the related method that now return time.Duration?

Reply:
durations are always in nansoseconds, so that makes sense to me. What do you think @jwair

Reply:
Agree that time.Duration is always in nanoseconds. For me that feels like an implementation detail of the time.Duration package rather than CTS. That is, when something is a time.Duration type (vs. a float/int), its no longer necessary for CTS to clarify that its in nanoseconds.

With regards to this constant name (vs. method names), I think the nanosecond naming could be confusing with the minutes unit on the right.

Reply:
yeah, I can get rid of the nanoseconds since the time package is handling all the conversions. I think it makes sense not to include it.